orderer/common/multichannel/registrar.go

```diff
@@ -446,3 +453,64 @@ func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
 	//TODO
 	return errors.New("Not implemented yet")
 }
+
+// validateJoinBlock returns whether this block can be used as a join block for the channel participation API
+// as well as whether the orderer is a member of the consenter set.
+// A join block is a block of either a system channel containing a ConsoritumsConfig, or an application channel
+// containing an ApplicationConfig.
+func (r *Registrar) validateJoinBlock(channelID string, configBlock *cb.Block) (isAppChannel bool, isMember bool, err error) {
+	if configBlock == nil {
+		return false, false, errors.New("nil block")
+	}
+
+	if configBlock.Data == nil || len(configBlock.Data.Data) == 0 {
+		return false, false, errors.New("empty block data")
+	}
+
+	firstTransaction := &cb.Envelope{}
+	if err := proto.Unmarshal(configBlock.Data.Data[0], firstTransaction); err != nil {
+		return false, false, errors.Wrap(err, "failed extracting envelope from block")
+	}
+
+	bundle, err := channelconfig.NewBundleFromEnvelope(firstTransaction, r.bccsp)
+	if err != nil {
+		return false, false, err
+	}
+
+	// Check channel type
+	_, isSystemChannel := bundle.ConsortiumsConfig()
+	_, isAppChannel = bundle.ApplicationConfig()
+
+	// A valid system channel has consortiums but no application group
+	// A valid application channel has an application group but no consortiums
+	if isSystemChannel == isAppChannel {
+		return false, false, errors.New("invalid channel type: must contain only one of consortiums or application group")
+	}
+
+	// Check if orderer is member of consenter set in join block
+	ordererConfig, ok := bundle.OrdererConfig()
+	if !ok {
+		return false, false, errors.New("channel config missing orderer configuration")
+	}
+
+	consensusType := ordererConfig.ConsensusType()
+	if consensusType != "etcdraft" {
+		return false, false, fmt.Errorf("join block consensus type is not configured to etcdraft, found type %s", consensusType)
+	}
+
+	etcdRaftMetadata := &eb.ConfigMetadata{}
+	err = proto.Unmarshal(ordererConfig.ConsensusMetadata(), etcdRaftMetadata)
+	if err != nil {
+		return false, false, errors.Wrap(err, "failed unmarshaling etcdraft metadata")
+	}
```

Comment:
Need to verify that the etcdRaftMetadata won't cause HandleChain to err, so better to use this as well:
etcdraft/utils.go CheckConfigMetadata() etc,